frontend/src/components/layout/AppPicker.tsx

```diff
@@ -0,0 +1,371 @@
+import {
+  useMenuTrigger,
+  useButton,
+  useMenu,
+  useOverlay,
+  FocusScope,
+  DismissButton,
+  mergeProps,
+  useMenuItem,
+  useFocus,
+} from "react-aria";
+import {
+  HTMLAttributes,
+  Key,
+  ReactNode,
+  useRef,
+  useState,
+  useEffect,
+} from "react";
+import { AriaMenuItemProps } from "@react-aria/menu";
+import { OverlayProps } from "@react-aria/overlays";
+import { useLocalization } from "@fluent/react";
+import { event as gaEvent } from "react-ga";
+import styles from "./AppPicker.module.scss";
+import BentoIconGrey from "../../../../static/images/bento.svg";
+import BentoIconWhite from "../../../../static/images/bento-white.svg";
+import FirefoxLogo from "../../../../static/images/logos/bento/fx.png";
+import MonitorLogo from "../../../../static/images/logos/bento/monitor.png";
+import PocketLogo from "../../../../static/images/logos/bento/pocket.png";
+import VpnLogo from "../../../../static/images/logos/bento/vpn.svg";
+import FxDesktopLogo from "../../../../static/images/logos/fx-logo.svg";
+import FxMobileLogo from "../../../../static/images/logos/bento/fx-mobile.png";
+import { Props as LayoutProps } from "./Layout";
+import { getRuntimeConfig } from "../../config";
+
+const getProducts = (referringSiteUrl: string) => ({
+  monitor: {
+    id: "monitor",
+    url: `https://monitor.firefox.com/?utm_source=${encodeURIComponent(
+      referringSiteUrl
+    )}&utm_medium=referral&utm_campaign=bento&utm_content=desktop`,
+  },
+  pocket: {
+    id: "pocket",
+    url: "https://app.adjust.com/hr2n0yz?engagement_type=fallback_click&fallback=https%3A%2F%2Fgetpocket.com%2Ffirefox_learnmore%3Fsrc%3Dff_bento&fallback_lp=https%3A%2F%2Fapps.apple.com%2Fapp%2Fpocket-save-read-grow%2Fid309601447",
+  },
+  fxDesktop: {
+    id: "fxDesktop",
+    url: `https://www.mozilla.org/firefox/new/?utm_source=${encodeURIComponent(
+      referringSiteUrl
+    )}&utm_medium=referral&utm_campaign=bento&utm_content=desktop`,
+  },
+  fxMobile: {
+    id: "fxMobile",
+    url: `https://www.mozilla.org/firefox/new/?utm_source=${encodeURIComponent(
+      referringSiteUrl
+    )}&utm_medium=referral&utm_campaign=bento&utm_content=desktop`,
+  },
+  vpn: {
+    id: "vpn",
+    url: `https://www.mozilla.org/products/vpn/?utm_source=${encodeURIComponent(
+      referringSiteUrl
+    )}&utm_medium=referral&utm_campaign=bento&utm_content=desktop`,
+  },
+});
+
+export const AppPicker = (props: { theme?: LayoutProps["theme"] } = {}) => {
+  const { l10n } = useLocalization();
+
+  const products = getProducts(getRuntimeConfig().frontendOrigin);
+  const monitorLinkRef = useRef<HTMLAnchorElement>(null);
+  const pocketLinkRef = useRef<HTMLAnchorElement>(null);
+  const fxDesktopLinkRef = useRef<HTMLAnchorElement>(null);
+  const fxMobileLinkRef = useRef<HTMLAnchorElement>(null);
+  const vpnLinkRef = useRef<HTMLAnchorElement>(null);
+  const mozillaLinkRef = useRef<HTMLAnchorElement>(null);
+
+  const onSelect = (itemKey: Key) => {
```

Comment:
Nice! Can't say that I've done anything specific for that, but I'll chalk that up as one of the advantages of settling on a standard widely used configuration :)

Reply:
Just figured out how to enable this for me: in VSCode, if you have the [vscode-jest](https://marketplace.visualstudio.com/items?itemName=Orta.vscode-jest) extension, it will usually just work by default. However, since this is a monorepo in which the frontend isn't at the root, I had to set the root path to `frontend` in my settings config:

```js
		"jest.rootPath": "frontend"
```

![image](https://user-images.githubusercontent.com/4251/152782258-25e6f5ba-3fc8-44a1-a9ae-cddb91587666.png)

Then it worked for me too :)

(I don't know how people feel about editor-specific configs in the repo, but I could add a file that makes this automatically work for anyone on VSCode+vscode-jest. Let me know.)

Reply:
🤚 I would love that file. I dont mind it.